### ISSUE TITLE
Symfony request created without any uri

### ISSUE THREAD

**Issue by aurimasniekis** [symfony/psr-http-message-bridge#16]
I tried to run symfony with this bridge on my http server and I was not able to leave homepage when I debugged Symfony request no uri is passed to request

$psrRequest->getUri();
```
object(Zend\Diactoros\Uri)#1452 (9) {
  ["allowedSchemes":protected]=>
  array(2) {
    ["http"]=>
    int(80)
    ["https"]=>
    int(443)
  }
  ["scheme":"Zend\Diactoros\Uri":private]=>
  string(4) "http"
  ["userInfo":"Zend\Diactoros\Uri":private]=>
  string(0) ""
  ["host":"Zend\Diactoros\Uri":private]=>
  string(14) "192.168.178.66"
  ["port":"Zend\Diactoros\Uri":private]=>
  int(1337)
  ["path":"Zend\Diactoros\Uri":private]=>
  string(12) "/favicon.ico"
  ["query":"Zend\Diactoros\Uri":private]=>
  string(0) ""
  ["fragment":"Zend\Diactoros\Uri":private]=>
  string(0) ""
  ["uriString":"Zend\Diactoros\Uri":private]=>
  NULL
}
```

$symfonyRequest
```
object(Symfony\Component\HttpFoundation\Request)#1524 (21) {
  ["attributes"]=>
  object(Symfony\Component\HttpFoundation\ParameterBag)#1527 (1) {
    ["parameters":protected]=>
    array(0) {
    }
  }
  ["request"]=>
  object(Symfony\Component\HttpFoundation\ParameterBag)#1525 (1) {
    ["parameters":protected]=>
    array(0) {
    }
  }
  ["query"]=>
  object(Symfony\Component\HttpFoundation\ParameterBag)#1526 (1) {
    ["parameters":protected]=>
    array(0) {
    }
  }
  ["server"]=>
  object(Symfony\Component\HttpFoundation\ServerBag)#1530 (1) {
    ["parameters":protected]=>
    array(0) {
    }
  }
  ["files"]=>
  object(Symfony\Component\HttpFoundation\FileBag)#1529 (1) {
    ["parameters":protected]=>
    array(0) {
    }
  }
  ["cookies"]=>
  object(Symfony\Component\HttpFoundation\ParameterBag)#1528 (1) {
    ["parameters":protected]=>
    array(0) {
    }
  }
  ["headers"]=>
  object(Symfony\Component\HttpFoundation\HeaderBag)#1531 (2) {
    ["headers":protected]=>
    array(7) {
      ["host"]=>
      array(1) {
        [0]=>
        string(19) "192.168.178.66:1337"
      }
      ["connection"]=>
      array(1) {
        [0]=>
        string(10) "keep-alive"
      }
      ["user-agent"]=>
      array(1) {
        [0]=>
        string(121) "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.111 Safari/537.36"
      }
      ["accept"]=>
      array(1) {
        [0]=>
        string(3) "*/*"
      }
      ["referer"]=>
      array(1) {
        [0]=>
        string(35) "http://192.168.178.66:1337/en/blog/"
      }
      ["accept-encoding"]=>
      array(1) {
        [0]=>
        string(19) "gzip, deflate, sdch"
      }
      ["accept-language"]=>
      array(1) {
        [0]=>
        string(14) "en-US,en;q=0.8"
      }
    }
    ["cacheControl":protected]=>
    array(0) {
    }
  }
  ["content":protected]=>
  string(0) ""
  ["languages":protected]=>
  NULL
  ["charsets":protected]=>
  NULL
  ["encodings":protected]=>
  NULL
  ["acceptableContentTypes":protected]=>
  NULL
  ["pathInfo":protected]=>
  NULL
  ["requestUri":protected]=>
  NULL
  ["baseUrl":protected]=>
  NULL
  ["basePath":protected]=>
  NULL
  ["method":protected]=>
  NULL
  ["format":protected]=>
  NULL
  ["session":protected]=>
  NULL
  ["locale":protected]=>
  NULL
  ["defaultLocale":protected]=>
  stri
```

**Comment by gcds** [comment]
Further investigation led me to conclusion that this bridge depends on PSR-7 Request created from global variables, and ignores the interface itself

**Comment by gcds** [comment]
My temporary hack was:

```php
$uri          = $psrRequest->getUri();
        $serverParams = [];
        if ('https' === $uri->getScheme()) {
            $serverParams['HTTPS'] = 'on';
        }

        $serverParams['SERVER_NAME']  = $uri->getHost();
        $serverParams['SERVER_PORT']  = $uri->getPort();
        $serverParams['REQUEST_URI']  = $uri->getPath();
        $serverParams['QUERY_STRING'] = $uri->getQuery();

        $serverParams = array_replace($serverParams, $psrRequest->getServerParams());

        $request = new Request(
            $psrRequest->getQueryParams(),
            $parsedBody,
            $psrRequest->getAttributes(),
            $psrRequest->getCookieParams(),
            $this->getFiles($psrRequest->getUploadedFiles()),
            $serverParams,
            $psrRequest->getBody()->__toString()
        );
```

**Comment by xabbuh** [comment]
Not sure I understand what you did and what went wrong. Can you provide some code that you have and then describe what you did expect and what happened instead?

**Comment by gcds** [comment]
This bridge implementation depends on global variables ($_SERVER) which some implementations fills PSR-7 ServerRequest serverParams. It should depend on ServerRequest Uri not ServerParams to create request url

**Comment by xabbuh** [comment]
The bridge does not make use of the `$_SERVER` supergobal. That's why I asked for some example code to reproduce your issue.

**Comment by gcds** [comment]
Here is the code:

```php
$httpFoundationFactory = new HttpFoundationFactory();
$symfonyRequest          = $httpFoundationFactory->createRequest($request);
```

https://github.com/symfony/psr-http-message-bridge/blob/master/Factory/HttpFoundationFactory.php#L33-L50

I don't see anythere in this code which uses PSR-7 ServerRequest getUri method. Inside Symfony Request it takes all URL related stuff from $_SERVER Params https://github.com/symfony/psr-http-message-bridge/blob/master/Factory/HttpFoundationFactory.php#L44 getServerParams from PSR-7 ServerRequest

**Comment by rougin** [comment]
Hello, same issue here. After creating a Symfony `Request` from `HttpFoundationFactory`, the factory did not get the URI from the PSR-compliant `Request` object. Calling the `Request::getPathInfo` method always returns a `/` string.

**Comment by gcds** [comment]
@rougin just use temp hack :)

**Comment by rougin** [comment]
Hi @gcds , thanks for the temporary fix! :+1:

**Comment by fabpot** [comment]
FYI, I've just released 0.4 with this patch.
